lib/Message.php

```diff
@@ -53,6 +53,9 @@ function getBodyAsStream() {
             rewind($stream);
             return $stream;
         }
+        if (is_callable($body)) {
+            throw new \UnexpectedValueException('Callback to stream not supported');
```

Comment:
This can also definitely be done, and would be nice to have! With output buffering this can be turned into a stream.